torchtune/datasets/_finetune.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Callable, Dict, Mapping, Optional
+
+import numpy as np
+
+from datasets import load_dataset
+from torch.utils.data import Dataset
+from torchtune.data import CROSS_ENTROPY_IGNORE_IDX, PromptTemplate
+from torchtune.modules.transforms import Transform
+
+
+class FinetuneDataset(Dataset):
```

Comment:
Is this actually only for SFT? If so, what makes it SFT specific? 

And if it is, then should we rename to `SFTDataset`????

Reply:
It is SFT specific in the sense that it isn't Preference or TextCompletion. i.e., it covers the previous instruct, chat, and eventual multimodal datasets but it does not cover 1) chosen/rejected messaged in Preference, 2) using `tokenizer.encode` directly instead of `tokenizer.tokenize_messages` in TextCompletion

I'm cool with SFTDataset tho

Reply:
Let's get some other opinions. 

@felipemello1 @SalmanMohammadi @pbontrager @ebsmothers

Reply:
m-m-m-m-milord? you asked for me?

SupervisedDataset? TaskDataset? SupervisedTaskDataset? STDataset? 

otherwise +1 for something slightly more descriptive than `FinetuneDataset` - not to be obtuse but, what else are you going to do with it?